dissect/cstruct/cstruct.py

```diff
@@ -196,37 +202,71 @@ def __getattr__(self, attr: str) -> Any:
         except KeyError:
             pass
 
+        try:
+            return self.types[attr]
+        except KeyError:
+            pass
+
         try:
             return self.resolve(self.typedefs[attr])
         except KeyError:
             pass
 
-        raise AttributeError(f"Invalid attribute: {attr}")
+        return super().__getattribute__(attr)
 
     def _next_anonymous(self) -> str:
         name = f"__anonymous_{self._anonymous_count}__"
         self._anonymous_count += 1
         return name
 
+    def _add_attr(self, name: str, value: Any, replace: bool = False) -> None:
+        if not replace and (name in self.__dict__ and self.__dict__[name] != value):
```

Comment:
```suggestion
        if not replace and (self.__dict__.get(name) != value):
```
Or is the overhead of the function call higher?